src/Plugins/RpcServer/RpcServer.Blockchain.cs

```diff
@@ -155,19 +137,19 @@ protected internal virtual JToken GetBlockHeader(JArray _params)
         /// <summary>
         /// Gets the state of a contract by its ID or script hash or (only for native contracts) by case-insensitive name.
         /// </summary>
-        /// <param name="_params">An array containing the contract ID or script hash or case-insensitive native contract name as the first element.</param>
+        /// <param name="contractNameOrHashOrId">Contract name or script hash or the native contract id.</param>
         /// <returns>The contract state in json format as a <see cref="JToken"/>.</returns>
-        [RpcMethod]
-        protected internal virtual JToken GetContractState(JArray _params)
+        [RpcMethodWithParams]
+        protected internal virtual JToken GetContractState(ContractNameOrHashOrId contractNameOrHashOrId)
         {
-            if (int.TryParse(_params[0].AsString(), out int contractId))
+            if (contractNameOrHashOrId.IsId)
             {
-                var contractState = NativeContract.ContractManagement.GetContractById(system.StoreView, contractId);
+                var contractState = NativeContract.ContractManagement.GetContractById(system.StoreView, contractNameOrHashOrId.AsId());
                 return contractState.NotNull_Or(RpcError.UnknownContract).ToJson();
             }
 
-            var scriptHash = Result.Ok_Or(() => ToScriptHash(_params[0].AsString()), RpcError.InvalidParams.WithData($"Invalid contract hash {_params[0]}"));
-            var contract = NativeContract.ContractManagement.GetContract(system.StoreView, scriptHash);
+            var hash = contractNameOrHashOrId.IsName ? ToScriptHash(contractNameOrHashOrId.AsName()) : contractNameOrHashOrId.AsHash();
```

Comment:
I see you treat the result of `.AsName()` as an address here. However, earlier this method was able to retrieve native contracts by name, and right now it can't. Not sure when this code was removed, but it also seems like a regression (and documentation confirms it).

Reply:
Agree, I missed that `ToScriptHash` handles it properly.